### PR TITLE
MH3: Arna Kennerüd, Skycaptain + 2 cards

### PR DESCRIPTION
**Pull request by MikeS-NZ**
MH3: Add card scripts for three MH3 cards, including token scripts:

- Arna Kennerüd, Skycaptain
- Genku, Future Shaper
- Thraben Charm
- 1/1 black Rat creature token with lifelink
- 1/2 blue Moonfolk creature token with flying
- 2/2 white Fox creature token with vigilance

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Name:Arna Kennerüd, Skycaptain
+ManaCost:2 W U B
+Types:Legendary Creature Human Knight
+PT:4/4
+K:Flying
+K:Lifelink
+K:Ward:Discard<1/Card>
+T:Mode$ Attacks | ValidCard$ Creature.modified+YouCtrl | TriggerZones$ Battlefield | Execute$ TrigDoubleCounters | TriggerDescription$ Whenever a modified creature you control attacks, double the number of each kind of counter on it. Then for each nontoken permanent attached to it, create a token that's a copy of that permanent attached to that creature.
+SVar:TrigDoubleCounters:DB$ MultiplyCounter | Defined$ TriggeredAttackerLKICopy | SubAbility$ DBCopyPermanents
+SVar:DBCopyPermanents:DB$ CopyPermanent | Defined$ Valid Permanent.nonToken+Attached | NumCopies$ 1 | AttachedTo$ TriggeredAttackerLKICopy
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt:10]: the valid looks incomplete?

File: forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Name:Genku, Future Shaper
+ManaCost:2 W U
+Types:Legendary Creature Moonfolk Wizard
+PT:2/5
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Any | ValidCard$ Permanent.YouCtrl+nonToken | TriggerZones$ Battlefield | Execute$ TrigCharm | TriggerDescription$ Whenever another nontoken permanent you control leaves the battlefield, choose one that hasn't been chosen this turn. Create a creature token with those characteristics.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt:5]: description incomplete?

File: forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+DeckHints:Type$Equipment|Aura 
+DeckHas:Ability$Token
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt:12]: |Discard|LifeGain

File: forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ManaCost:2 W U B
+Types:Legendary Creature Human Knight
+PT:4/4
+K:Flying
+K:Lifelink
+K:Ward:Discard<1/Card>
+T:Mode$ Attacks | ValidCard$ Creature.modified+YouCtrl | TriggerZones$ Battlefield | Execute$ TrigDoubleCounters | TriggerDescription$ Whenever a modified creature you control attacks, double the number of each kind of counter on it. Then for each nontoken permanent attached to it, create a token that's a copy of that permanent attached to that creature.
+SVar:TrigDoubleCounters:DB$ MultiplyCounter | Defined$ TriggeredAttackerLKICopy | SubAbility$ DBCopyPermanents
+SVar:DBCopyPermanents:DB$ CopyPermanent | Defined$ Valid Permanent.nonToken+Attached | NumCopies$ 1 | AttachedTo$ TriggeredAttackerLKICopy
+DeckHints:Type$Equipment|Aura
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt:11]: |Counters

File: forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+SVar:TrigCharm:DB$ Charm | ChoiceRestriction$ ThisTurn | Choices$ DBToken1,DBToken2,DBToken3
+SVar:DBToken1:DB$ Token | TokenAmount$ 1 | TokenScript$ w_2_2_fox_vigilance | TokenOwner$ You | SpellDescription$ 2/2 white Fox with vigilance.
+SVar:DBToken2:DB$ Token | TokenAmount$ 1 | TokenScript$ u_1_2_moonfolk_flying | TokenOwner$ You | SpellDescription$ 1/2 blue Moonfolk with flying.
+SVar:DBToken3:DB$ Token | TokenAmount$ 1 | TokenScript$ b_1_1_rat_lifelink | TokenOwner$ You | SpellDescription$ 1/1 black Rat with lifelink.
+A:AB$ PutCounterAll | Cost$ 3 W U | ValidCards$ Creature.YouCtrl | ValidCardsDesc$ creature you control | CounterType$ P1P1 | CounterNum$ 1 | SpellDescription$ Put a +1/+1 counter on each creature you control.
+DeckHas:Ability$Counters|Token
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt:11]: |LifeGain & Types$Fox|Moonfolk|Rat & Color$Black

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt:11]: Actually the Type part might not be needed thanks to tool4ever latest PR

File: forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt (hunk 1 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt:10]: I've given this a lot of testing and I'm not sure how to approach the valid differently. I see sometimes the copying isn't functioning right, but not sure of the fix

File: forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt (hunk 2 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt:5]: This is the same description format as Outlaws' Merriment I think. The token descriptions appear on the `Token` lines

File: forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt (hunk 5 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt]: Added `LifeGain` ability tag

File: forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt (hunk 3 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt]: Added counters and lifegain tag

File: forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt:5]: It seems incomplete
When in doubt it's always good to confirm ingame if everything shows up:
grafik

File: forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt:10]: Well Attached property alone has no additional context, so it just assumes you're checking against the trait host

check e.g. _Rhuk, Hexgold Nabber_

Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/arna_kennerud_skycaptain.txt]: Okay, thank you for the tip! I reworked the `Defined$` parameter to use the `AttachedTo` tag, it seems to work with all valid creatures in testing. I'm still not sure whether the `TriggeredAttackerLKICopy` is correct per the oracle text, but `TriggeredCardLKICopy` didn't work in its place

File: forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt (hunk 2 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt:5]: I see the full list of tokens isn't being included in the card's hover-over text. I can't for the life of me figure out why this is the case when Outlaws' Merriment appears to not have this problem, but is scripted the same way? (I.e. the list of tokens isn't hard-baked into the trigger description.)

If I copy-paste the `\n• 2/2 white Fox with vigilance.\n• 1/2 blue Moonfolk with flying.\n• 1/1 black Rat with lifelink.` text into the trigger description, the hover-over text reflects this and looks fine, but the trigger description on the stack itself starts to look very ugly (the newline esc characters appear in the raw text). Is this only solvable with a unique `StackDescription` or am I approaching this wrong?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt:5]: you need to use ABILITY syntax to tell the engine to include the charm description

**Review by tehdiplomat — APPROVED** [review]
Gonna merge so we can fix the missing description after merge

File: forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt (hunk 2 above)
Comment (tehdiplomat) [reply forge-gui/res/cardsfolder/upcoming/genku_future_shaper.txt:5]: Will fix after merge

**Pull request merged**